src/networking.c

```diff
@@ -2014,6 +2015,7 @@ void resetClient(client *c) {
     c->multibulklen = 0;
     c->bulklen = -1;
     c->slot = -1;
+    c->duration = 0;
```

Comment:
good idea to put it here, however, i think there's a problem with scripts.
luaRedisGenericCommand only calls freeClientArgv.
maybe we should change luaRedisGenericCommand to call resetClient, which also does a few excess steps, or maybe this reset of c->duration belongs in freeClientArgv? the function name doesn't seem like a match, but it's actions (also resetting c->cmd) could be an ok fit.
i'm leaning towards freeClientArgv.

either way, let's try to add a test that can detect this problem.

Reply:
@oranagra  I understand your point regarding LUA scripts. However I do not think placing it in freeClientArgv is the correct solution.
The reason is that IMO It is logically wrong to assume this is were we finish/start processing new command.
For example I think blocking zpop, when reprocessed might call rewriteClientCommandVector which will free the argv.
So in this case in my implementation we will loose the duration of the first  processing try.

IMO since LUA is already doing a "proprietary" handling of preparing the client for the next command , probably since it cannot really call resetClient as it will also reset some other things which might be needed during the LUA execution (ASKING flag, pointer to cur_script etc..) , It should explicitly zero the duration.

Reply:
good point, but then, maybe we should split the role of resetClient into two, or pass some boolean argument?
i.d rather put the logic in one place and have script_lua.c call it than clone individual lines (increases the chance something else will be overlooked one day)

Reply:
@oranagra I did something  :) take a look if that is what you meant

Reply:
i looked at the change, it bothered me that the documentation of the function says what it does (lists the fields it zeros), but not what it's purpose.
i'm wondering again why we can't change luaRedisGenericCommand to simply call resetClient?

Reply:
It bothered me 2 as I told you I have hard time trying to understand what we encapsulate. I think there might be issues reseting client during lua script. for example do we want to clear the ASKING flag (this is possible to run LUA after ASKING right?) also the current slot is zeroed and the request type which are some examples of things we might not to reset IMO.
Maybe I am wrong but LUA solves it by clearing some fields itself.
I can maybe suggest that we can create a separate function to resetClient during LUA?

Reply:
in lua scripts, the client which we reset is the lua (internal) client, not the real client that called the script.

Reply:
Oh. So you say it will be O.K to reset the semi client. 
I added the resetClient instead. lets see if it fails any tests